src/Microsoft.AspNet.Razor/Generator/Compiler/CodeBuilder/CSharp/Visitors/CSharpCodeVisitor.cs

```diff
@@ -500,17 +500,23 @@ private CSharpCodeWriter RenderPreWriteStart()
             return RenderPreWriteStart(Writer, Context);
         }
 
-        public static CSharpCodeWriter RenderPreWriteStart(CSharpCodeWriter writer, CodeBuilderContext context)
+        public static CSharpCodeWriter RenderPreWriteStart(
+            CSharpCodeWriter writer, CodeBuilderContext context, bool isTagHelperOutput = false)
         {
+            string writeMethodName;
             if (!string.IsNullOrEmpty(context.TargetWriterName))
             {
-                writer.WriteStartMethodInvocation(context.Host.GeneratedClassContext.WriteLiteralToMethodName)
+                writeMethodName = isTagHelperOutput ? context.Host.GeneratedClassContext.WriteMethodName
+                    : context.Host.GeneratedClassContext.WriteLiteralToMethodName;
```

Comment:
if you have to wrap a ternary, wrap all 3 parts.  either
```
  writeMethodName = isTagHelperOutput ?
    context.Host.GeneratedClassContext.WriteMethodName :
    context.Host.GeneratedClassContext.WriteLiteralToMethodName;
```
or just use an `if` since the ternary isn't helping much.